File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_publisher/lib/src/services/publisher_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/publisher_models.dart';

/// Publisher service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class PublisherService {
  const PublisherService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'publisher';

  PublisherPageResult<Never> empty() =>
      const PublisherPageResult<Never>(items: <Never>[]);
}

File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-publisher/src/services/publisherService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { PublisherPageResult } from "../types/publisherModels.js";

/**
 * Publisher service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface PublisherService {
  readonly capability: "publisher";
}

export function createPublisherService(client: AppstoreAppClient): PublisherService {
  void client;
  return { capability: "publisher" };
}

export function emptyPublisherResult(): PublisherPageResult<never> {
  return { items: [] };
}
